sdk/src/authenticate.tsx

```diff
@@ -0,0 +1,466 @@
+     *
+     * @return {Promise<string>} - A Promise that resolves with the id token.
+     *
+     * @example
+     * ```
+     * const idToken = await getIDToken();
+     * ```
+     */
+    const getIDToken = async (): Promise<string> => {
+
+        return await auth.getIDToken();
+    }
+
+    /**
+     * This method returns if the user is authenticated or not.
+     *
+     * @return {Promise<boolean>} - A Promise that resolves with `true` if the user is authenticated, `false` otherwise.
+     *
+     * @example
+     * ```
+     * await isAuthenticated();
+     * ```
+     */
+    const isAuthenticated = async (): Promise<boolean> => {
+
+        return await auth.isAuthenticated();
+    }
+
+    /**
+     * This method returns the PKCE code generated during the generation of the authentication URL.
+     *
+     * @return {Promise<string>} - A Promise that resolves with the PKCE code.
+     *
+     * @example
+     * ```
+     * const pkce = await getPKCECode();
+     * ```
+     */
+    const getPKCECode = async (): Promise<string> => {
+
+        return await auth.getPKCECode();
+    }
+
+    /**
+     * This method sets the PKCE code to the data store.
+     *
+     * @param {string} pkce - The PKCE code.
+     * @return  {Promise<void>}
+     * 
+     * @example
+     * ```
+     * await setPKCECode("pkce_code")
+     * ```
+     */
+    const setPKCECode = async (pkce: string): Promise<void> => {
+
+        return await auth.setPKCECode(pkce);
+    }
+
+    /**
+     * This function will handle authentication/ signout redirections.
+     * 
+     * @param {AuthUrl} authUrl - redirection url.
+     * @return  {Promise<void>}
+     */
+    const handleAuthRedirect = async (authUrl: AuthUrl): Promise<void> => {
+        try {
+            if (url.parse(authUrl?.url)?.query.indexOf("code=") > -1) {
+                await requestAccessTokenDetails(authUrl);
+            } else if (url.parse(authUrl?.url)?.query.indexOf("state=sign_out") > -1) {
+                const dataList = url.parse(authUrl?.url)?.query.split("&");
+                const authState = dataList[0].split("=")[1];
+        
+                if (authState === "sign_out_success") {
+                    try {
+                        await auth.getDataLayer().removeOIDCProviderMetaData();
+                        await auth.getDataLayer().removeTemporaryData();
+                        await auth.getDataLayer().removeSessionData();
+                        setState(initialState);
+                    } catch(error) {
+                        throw new AsgardeoAuthException(
+                            "AUTHENTICATE-HAR-IV01",
+                            "authenticate",
+                            "handleAuthRedirect",
+                            "Error in signout",
+                            error
+                        )
+                    }
+                }
+            } else {
+                console.log("Invalid redirection url");
```

Comment:
I think it is better to remove all console logs

Reply:
Removed and added `TODO`s to track logging as a future development.